core/plugins/org.polarsys.capella.core.ui.search/src/org/polarsys/capella/core/ui/search/CapellaSearchSettings.java

```diff
@@ -118,23 +118,28 @@ public void setAbstractChecked(boolean isAbstractChecked) {
   public void setSemanticChecked(boolean isSemanticChecked) {
     this.isSemanticChecked = isSemanticChecked;
   }
-  
+
   public void setScope(int scope) {
     this.scope = scope;
   }
-  
+
   public int getScope() {
     return this.scope;
   }
 
-  // method used to check the search settings (that we entered text, selected at least one mettaclass or attribute etc) 
+  // method used to check the search settings (that we entered text, selected at least one mettaclass or attribute etc)
   public IStatus validate() {
     if (textPattern == null || textPattern.isEmpty()) {
       return new Status(IStatus.ERROR, Activator.PLUGIN_ID,
           CapellaSearchConstants.CapellaSearchPage_Validation_Message_Pattern_Empty);
     }
-    
+
     if (isRegExSearch) {
+      if (isWholeWord) {
+        return new Status(IStatus.ERROR, Activator.PLUGIN_ID,
+            CapellaSearchConstants.CapellaSearchPage_Validation_Message_Whole_Word_Same_Time_Regex);
+      }
+

```

Comment:
This is required since `org.eclipse.search.internal.core.text.PatternConstructor.createPattern(String, boolean, boolean, boolean, boolean)` throws an exception otherwise.